go-controller/pkg/ovn/namespace.go

```diff
@@ -409,8 +410,38 @@ func (oc *Controller) deleteNamespaceLocked(ns string) *namespaceInfo {
 }
 
 func (oc *Controller) createNamespaceAddrSetAllPods(ns string) (addressset.AddressSet, error) {
-	// Get all the pods in the namespace and append their IP to the address_set
 	var ips []net.IP
+	// special handling of host network namespace
+	if config.Kubernetes.HostNetworkNamespace != "" &&
+		ns == config.Kubernetes.HostNetworkNamespace {
+		// add the mp0 interface addresses to this namespace.
+		existingNodes, err := oc.watchFactory.GetNodes()
+		if err != nil {
+			klog.Errorf("Failed to get all nodes (%v)", err)
+		} else {
+			ips = make([]net.IP, 0, len(existingNodes))
+			for _, node := range existingNodes {
+				hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node)
+				if err != nil {
+					klog.Errorf("Error parsing host subnet annotation for node %s (%v)",
+						node.Name, err)
+					continue
+				}
+				for _, hostSubnet := range hostSubnets {
```

Comment:
can we have multiple managment IP / node ?

Reply:
ovn-k has a concept of single mp0 interface per subnet that is assigned to the node. It is possible to have multiple subnets per node.